src/command/ping-command.ts

```diff
@@ -47,12 +47,20 @@ export class PingCommand implements CommandInterface<PingOptions> {
 			});
 		});
 
-		const result = await cmd;
+		let result = {};
+		try {
+			const cmdResult = await cmd;
+			result = this.parse(cmdResult.stdout);
+		} catch (error: unknown) {
+			result = {
+				rawOutput: (error as ExecaError).stderr.toString(),
```

Comment:
The reason for `catch(error: unknown)` is to force developers to explicitly check the error types. In that case, it could be an execa error OR an exception from `this.parse` call. In the second case, the error won't have `stderr` property.

Reply:
The problem is, `ExecaError` is not a class, its a type. Execa returns a plain object error with custom variables. `instanceof` check will fail. Best I could do is to check for the `stderr` value.

Reply:
this should do the trick.

Reply:
[Type guard technique](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) can be used in such cases.

```
const isExecaError = (error: any): error is ExecaError => {
    return (<ExecaError>error).stderr !== undefined;
};
```

But no need to complicate it for now.